src/dsymbol.d

```diff
@@ -392,13 +392,13 @@ extern (C++) class Dsymbol : RootObject
         return null;
     }
 
-    final Dsymbol pastMixin()
+    final inout(Dsymbol) pastMixin() inout
     {
-        Dsymbol s = this;
+        Dsymbol s = cast()this;
         //printf("Dsymbol::pastMixin() %s\n", toChars());
         while (s && s.isTemplateMixin())
             s = s.parent;
-        return s;
+        return cast(inout(Dsymbol))s;
     }
```

Comment:
Time to eat our dogfood. TDPL says black on white we do tail call optimization. A variety of public online resources advertise the same. This function must be:

```d
final inout(Dsymbol) pastMixin() inout
{
    if (!isTemplateMixin) return this;
    if (!parent) return null;
    return parent.pastMixin;
}
```

Thanks.